docs/content/modules/carts-and-checkout/backend/add-payment-provider.md

```diff
@@ -165,41 +174,46 @@ class MyPaymentService extends AbstractPaymentService {
 }
 ```
 
-### createPayment
+### PaymentProcessorError
+
+Before diving into the methods you'll need to implement, you'll notice that part of the expected return signature of these method includes `PaymentProcessorError`. This is an interface of the following definition:
+
+```ts
+interface PaymentProcessorError {
+  error: string
+  code?: string
+  detail?: any
+}
+```
+
+While implementing the following methods, if you need to inform the Medusa core that an error occurred at a certain stage, return an object having the attributes defined in the `PaymentProcessorError` interface.
+
+### initiatePayment
 
 This method is called during checkout when [Payment Sessions are initialized](/api/store/#tag/Cart/operation/PostCartsCartPaymentSessions) to present payment options to the customer. It is used to allow you to make any necessary calls to the third-party provider to initialize the payment.
 
-For example, in Stripe this method is used to initialize a Payment Intent for the customer.
+For example, in Stripe this method is used to create a Payment Intent for the customer.
```

Comment:
for line 193: Todo: This is not called when payment sessions are initialised (except when there is only one provider available in the region). this occurs when the payment session is set as active. see flow sub issue